pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/mapper/HadoopSegmentCreationMapReduceJob.java

```diff
@@ -191,26 +199,48 @@ protected void map(LongWritable key, Text value, Context context) throws IOExcep
 
       try {
         String segmentName =
-            createSegment(_inputFilePath, schema, Integer.parseInt(lineSplits[2]), hdfsInputFilePath, localInputDataDir,
-                fs);
+                createSegment(_inputFilePath, schema, Integer.parseInt(lineSplits[2]), hdfsInputFilePath, localInputDataDir,
+                        fs);
         LOGGER.info(segmentName);
         LOGGER.info("Finished segment creation job successfully");
       } catch (Exception e) {
         LOGGER.error("Got exceptions during creating segments!", e);
       }
 
       context.write(new LongWritable(Long.parseLong(lineSplits[2])), new Text(
-          FileSystem.get(_properties).listStatus(new Path(_localHdfsSegmentTarPath + "/"))[0].getPath().getName()));
+              FileSystem.get(_properties).listStatus(new Path(_localHdfsSegmentTarPath + "/"))[0].getPath().getName()));
       LOGGER.info("Finished the job successfully");
     }
 
     protected void setSegmentNameGenerator(SegmentGeneratorConfig segmentGeneratorConfig, Integer seqId,
-        Path hdfsAvroPath, File dataPath) {
+                                           Path hdfsAvroPath, File dataPath) {
+      if(segmentGeneratorConfig.getSchema() == null ||
+              segmentGeneratorConfig.getSchema().getTimeFieldSpec() == null ||
+              segmentGeneratorConfig.getSchema().getTimeFieldSpec().getIncomingGranularitySpec() == null){
```

Comment:
could we use outgoing granularity spec?

Reply:
what happens to the case where there is only incoming granularity ?

Reply:
@Nonnull
  public TimeGranularitySpec getOutgoingGranularitySpec() {
    if (_outgoingGranularitySpec == null) {
      return _incomingGranularitySpec;
    } else {
      return _outgoingGranularitySpec;
    }
  }